File: benchmarks/primes/primes.rs
fn main() {
    let n = 3_000_000;
    let mut sieve = vec![false; n as usize + 1];
    let mut count = 0;
    for i in 2..=n {
        if !sieve[i as usize] {
            count += 1;
            let mut j = i + i;
            while j <= n {
                sieve[j as usize] = true;
                j += i;
            }
        }
    }
    println!("Primes: {}", count);
}

File: benchmarks/primes/primes.go
package main

func main() {
	n := 3000000
	sieve := make([]bool, n+1)
	count := 0
	for i := 2; i <= n; i++ {
		if !sieve[i] {
			count++
			for j := i + i; j <= n; j += i {
				sieve[j] = true
			}
		}
	}
	println("Primes:", count)
}
